service/qos/service_level_controller.cc

```diff
@@ -240,10 +246,17 @@ future<> service_level_controller::notify_service_level_updated(sstring name, se
                 try {
                     subscriber->on_before_service_level_change(slo_before, slo, {name}).get();
                 } catch (...) {
-                    sl_logger.error("notify_service_level_updated: exception occurred in one of the observers callbacks {}", std::current_exception());
+                    sl_logger.error("notify_before_service_level_updated: exception occurred in one of the observers callbacks {}", std::current_exception());
                 }
             });
             sl_it->second.slo = slo;
+            _subscribers.thread_for_each([name, slo_before, slo] (qos_configuration_change_subscriber* subscriber) {
+                try {
+                    subscriber->on_after_service_level_change(slo_before, slo, {name}).get();
+                } catch (...) {
```

Comment:
Why do we need a thread here?

I guess you just copied the other function. Please follow up later with de-threading.